modules/seqexec/server/src/main/scala/seqexec/server/tcs/Tcs.scala

```diff
@@ -102,16 +105,16 @@ final case class Tcs private (tcsController: TcsController,
             calcGuiderInUse(c.tcsGuide, TipTiltSource.PWFS1, M1Source.PWFS1) | aoUsesP1,
             config.guideWithP1)
           ),
-          tag[P2Config](calcGuiderConfig(
-            calcGuiderInUse(c.tcsGuide, TipTiltSource.PWFS2, M1Source.PWFS2),
-            config.guideWithP2)
+          usesAltair.either(
+            tag[P2Config](calcGuiderConfig(calcGuiderInUse(c.tcsGuide, TipTiltSource.PWFS2, M1Source.PWFS2),
+              config.guideWithP2)),
+            tag[AoGuide](calcGuiderConfig(calcGuiderInUse(c.tcsGuide, TipTiltSource.GAOS, M1Source.GAOS),
+              config.guideWithAO).tracking)
```

Comment:
I'm not familiar with shapeless yet, so I'll defer to @swalker2m to provide any feedback here if he has any.